src/engraving/libmscore/score.cpp

```diff
@@ -2617,7 +2617,7 @@ void Score::removePart(Part* part)
 //   insertStaff
 //---------------------------------------------------------
 
-void Score::insertStaff(Staff* staff, staff_idx_t ridx)
+void Score::insertStaff(Staff* staff, staff_idx_t ridx, bool systemObjectStaff)
```

Comment:
What about Score::appendStaff(Staff* staff)? Either this flag should not be here (and we will call score->addSystemObjectStaff explicitly), or you should also add it to Score::addStaff